export/export.go

```diff
@@ -0,0 +1,296 @@
+		}
+		resources.Pvcs = pvcs.Items
+		return nil
+	})
+
+	if err := sem.Acquire(ctx, 1); err != nil {
+		return nil, xerrors.Errorf("acquire semaphore: %w", err)
+	}
+	g.Go(func() error {
+		defer sem.Release(1)
+		scs, err := s.storageClassService.List(ctx)
+		if err != nil {
+			return xerrors.Errorf("to call list storageClasses")
+		}
+		resources.StorageClasses = scs.Items
+		return nil
+	})
+
+	if err := sem.Acquire(ctx, 1); err != nil {
+		return nil, xerrors.Errorf("acquire semaphore: %w", err)
+	}
+	g.Go(func() error {
+		defer sem.Release(1)
+		ss := s.schedulerService.GetSchedulerConfig()
+		resources.SchedulerConfig = ss
+		return nil
+	})
+
+	if err := g.Wait(); err != nil {
+		return nil, xerrors.Errorf("get resources all: %w", err)
+	}
+	return &resources, nil
+}
+
+func (s *Service) Export(ctx context.Context) (*Resources, error) {
+	resources, err := s.get(ctx)
+	if err != nil {
+		return nil, xerrors.Errorf("export resources all: %w", err)
+	}
+	return resources, nil
+}
+
+// Import all resources from posted data.
+// (1) Restart scheduler based on the data.
+// (2) Apply each resource to the scheduler.
+//     * If UID is not nil, an error will occur. (try to find existing resource by UID)
+// (3) Get all resources. (Separated the get function to unify the struct format.)
+func (s *Service) Import(ctx context.Context, resources *ResourcesApplyConfiguration) error {
+	// TODO: Issue: #12 PR: #13
+	// if err := s.schedulerService.RestartScheduler(resources.SchedulerConfig); err != nil {
+	// 	klog.Warningf("failed to start scheduler with imported configuration: %v", err)
+	// 	return nil, xerrors.Errorf("restart scheduler with imported configuration: %w", err)
+	// }
+	g, _ := errgroup.WithContext(ctx)
+	sem := semaphore.NewWeighted(int64(runtime.GOMAXPROCS(0)))
+
+	for i := range resources.StorageClasses {
+		sc := resources.StorageClasses[i]
+		if err := sem.Acquire(ctx, 1); err != nil {
+			return xerrors.Errorf("acquire semaphore: %w", err)
+		}
+		g.Go(func() error {
+			defer sem.Release(1)
+			sc.ObjectMetaApplyConfiguration.UID = nil
+			_, err := s.storageClassService.Apply(ctx, &sc)
+			if err != nil {
+				return xerrors.Errorf("apply StorageClass: %w", err)
+			}
+			return nil
+		})
+
+	}
+	for i := range resources.Pvcs {
+		pvc := resources.Pvcs[i]
+		if err := sem.Acquire(ctx, 1); err != nil {
+			return xerrors.Errorf("acquire semaphore: %w", err)
+		}
+		g.Go(func() error {
+			defer sem.Release(1)
+			pvc.ObjectMetaApplyConfiguration.UID = nil
+			_, err := s.pvcService.Apply(ctx, &pvc)
+			if err != nil {
+				return xerrors.Errorf("apply PersistentVolumeClaims: %w", err)
+			}
+			return nil
+		})
+	}
+	for i := range resources.Pvs {
+		pv := resources.Pvs[i]
+		if err := sem.Acquire(ctx, 1); err != nil {
+			return xerrors.Errorf("acquire semaphore: %w", err)
+		}
+		g.Go(func() error {
+			defer sem.Release(1)
+			pv.ObjectMetaApplyConfiguration.UID = nil
+			if *pv.Status.Phase == "Bound" {
```

Comment:
`pv.Status.Phase` is set option to `omitempty`. Therefore, if it is nil, it happens nil pointer error.
In general, no need to be afraid of that. But it may happen if someone rewrites it manually.
How do we check the parameter `Phase` of the struct is exists?

Reply:
How about this:
```go
if pv.Status.Phase != nil {
}
```

Reply:
oops, why couldn't I...thank you